packages/contracts/src/contracts/current/protocol/Exchange/MixinMatchOrders.sol

```diff
@@ -0,0 +1,315 @@
+        uint256 amountSpentByLeft = safeAdd(
+            matchedFillResults.right.takerAssetFilledAmount,
+            matchedFillResults.takerFillAmount
+        );
+        require(
+            matchedFillResults.left.makerAssetFilledAmount >=
+            amountSpentByLeft,
+            MISCALCULATED_TRANSFER_AMOUNTS
+        );
+        // If the amount transferred from the left order is different than what is transferred, it is a rounding error amount.
+        // Ensure this difference is negligible by dividing the values with each other. The result should equal to ~1.
+        require(
+            !isRoundingError(
+                matchedFillResults.left.makerAssetFilledAmount,
+                amountSpentByLeft,
+                1
+            ),
+            ROUNDING_ERROR_TRANSFER_AMOUNTS
+        );
+
+        // The right order must spend at least as much as we're transferring to the left order.
+        require(
+            matchedFillResults.right.makerAssetFilledAmount >=
+            matchedFillResults.left.takerAssetFilledAmount,
+            MISCALCULATED_TRANSFER_AMOUNTS
+        );
+        // If the amount transferred from the right order is different than what is transferred, it is a rounding error amount.
+        // Ensure this difference is negligible by dividing the values with each other. The result should equal to ~1.
+        require(
+            !isRoundingError(
+                matchedFillResults.right.makerAssetFilledAmount,
+                matchedFillResults.left.takerAssetFilledAmount,
+                1
+            ),
+            ROUNDING_ERROR_TRANSFER_AMOUNTS
+        );
+    }
+
+    /// @dev Calculates fill amounts for the matched orders.
+    ///      Each order is filled at their respective price point. However, the calculations are
+    ///      carried out as though the orders are both being filled at the right order's price point.
+    ///      The profit made by the leftOrder order goes to the taker (who matched the two orders).
+    /// @param leftOrder First order to match.
+    /// @param rightOrder Second order to match.
+    /// @param leftOrderStatus Order status of left order.
+    /// @param rightOrderStatus Order status of right order.
+    /// @param leftOrderFilledAmount Amount of left order already filled.
+    /// @param rightOrderFilledAmount Amount of right order already filled.
+    /// @param matchedFillResults Amounts to fill and fees to pay by maker and taker of matched orders.
+    function calculateMatchedFillResults(
+        Order memory leftOrder,
+        Order memory rightOrder,
+        uint8 leftOrderStatus,
+        uint8 rightOrderStatus,
+        uint256 leftOrderFilledAmount,
+        uint256 rightOrderFilledAmount
+    )
+        internal
+        returns (MatchedFillResults memory matchedFillResults)
+    {
+        // We settle orders at the price point defined by the right order (profit goes to the order taker)
```

Comment:
> we are finding the minimum remaining fillable amounts given two orders

We're comparing the fillable amount of the left order with the spendable amount of the right order.
Another way to phrase it would be: If right can sell more than left can buy, then right fills left.

I've updated this comment for clarity, but certainly let me know if the wording can still be improved.